modules/spring-config/src/test/java/org/mule/runtime/config/internal/registry/SpringRegistryTestCase.java

```diff
@@ -78,12 +82,13 @@ public void unregisterBeanWhoseCreationFails() throws RegistrationException {
                                                  mock(MuleContext.class), mock(ConfigurationDependencyResolver.class),
                                                  mock(LifecycleInterceptor.class));
 
-    when(beanFactory.containsBeanDefinition("key")).thenReturn(true);
+    beanFactory.registerBeanDefinition("key", new GenericBeanDefinition());
     when(appContext.getBean("key")).thenThrow(BeanCreationException.class);
 
     assertThat(registry.unregisterObject("key"), is(nullValue()));
-    verify(beanFactory).removeBeanDefinition("key");
-    verify(beanFactory).destroySingleton("key");
+    assertThrows("No bean named 'key' available", NoSuchBeanDefinitionException.class,
```

Comment:
Just a warning here: the first argument of assertThrows is the message to be printed if the assertion isn't success. If you want to assert the exception error message, it would be:
```
Exception thrown = assertThrows("No bean named 'key' available", NoSuchBeanDefinitionException.class, ...);
// do assertions over "thrown"
```